cockatrice/src/window_main.cpp

```diff
@@ -200,7 +200,15 @@ void MainWindow::actConnect()
 
 void MainWindow::actRegister()
 {
-    DlgRegister dlg(this);
+    // Note the pre-existing form data so that if the user is bounced back to the dialogue, they
+    // won't have to re-enter all the information again.
+    QString username = client->getUserName();
+    QString password = client->password;
+    QString email = client->email;
+    QString country = client->country;
+    QString realname = client->realName;
```

Comment:
@ebbit1q now `MainWindow` is the `friend class` instead of `DlgRegister`, and these variables are passed to the `DlgRegister` constructor